apps/somnia/src/commands/archive.ts

```diff
@@ -0,0 +1,263 @@
+  StringSelectMenuInteraction,
+} from 'discord.js'
+import { SlashCommandBuilder } from 'discord.js'
+import * as fs from 'fs'
+import * as path from 'path'
+
+import { error } from '../lib/message'
+import { cwd } from '../util'
+
+import {
+  ArtifactSetKey,
+  LocationCharacterKey,
+  WeaponKey,
+} from '@genshin-optimizer/gi/consts'
+import { artifactarchive } from './archive/artifact'
+import { chararchive } from './archive/char'
+import { weaponarchive } from './archive/weapon'
+
+export const slashcommand = new SlashCommandBuilder()
+  .setName('archive')
+  .setDescription('Genshin Archive')
+  .addSubcommand((s) =>
+    s
+      .setName('char')
+      .setDescription('Characters')
+      .addStringOption((o) =>
+        o
+          .setName('name')
+          .setDescription('Character name')
+          .setAutocomplete(true)
+          .setRequired(true)
+      )
+      .addStringOption((o) =>
+        o.setName('talent').setDescription('Talent name').setAutocomplete(true)
+      )
+  )
+  .addSubcommand((s) =>
+    s
+      .setName('weapon')
+      .setDescription('Weapons')
+      .addStringOption((o) =>
+        o
+          .setName('name')
+          .setDescription('Weapon name')
+          .setAutocomplete(true)
+          .setRequired(true)
+      )
+      .addIntegerOption((o) =>
+        o
+          .setName('refine')
+          .setDescription('Refinement level (1-5)')
+          .setMinValue(1)
+          .setMaxValue(5)
+      )
+  )
+  .addSubcommand((s) =>
+    s
+      .setName('artifact')
+      .setDescription('Artifacts')
+      .addStringOption((o) =>
+        o
+          .setName('name')
+          .setDescription('Artifact set name')
+          .setAutocomplete(true)
+          .setRequired(true)
+      )
+  )
+
+//requiring all the data because imports dont work
+const archivepath = path.join(cwd, '/libs/gi/dm-localization/assets/locales/en')
+//get keys
+const archive: Record<string, any> = {
+  key: {
+    char: require(path.join(archivepath, '/charNames_gen.json')),
+    weapon: require(path.join(archivepath, '/weaponNames_gen.json')),
+    artifact: require(path.join(archivepath, '/artifactNames_gen.json')),
+  },
+  char: {},
+  weapon: {},
+  artifact: {},
+}
+//traveler data
+for (const name in archive['key']['char']) {
+  if (name.match(/Traveler/)) delete archive['key']['char'][name]
+}
+archive['key']['char']['TravelerAnemoF'] = 'Traveler (Anemo)'
+archive['key']['char']['TravelerGeoF'] = 'Traveler (Geo)'
+archive['key']['char']['TravelerElectroF'] = 'Traveler (Electro)'
+archive['key']['char']['TravelerDendroF'] = 'Traveler (Dendro)'
+archive['key']['char']['TravelerHydroF'] = 'Traveler (Hydro)'
+//get all the data from keys
+for (const category in archive['key']) {
+  for (const name in archive['key'][category]) {
+    const itempath = path.join(archivepath, `/${category}_${name}_gen.json`)
+    if (fs.existsSync(itempath)) archive[category][name] = require(itempath)
+  }
+}
+
+const colors = {
```

Comment:
These colors and images should exist somewhere in our libraries. Tho they might be SVG, not sure if embeds support that

Reply:
embeds don't actually take files, only links. so i need an image link

Reply:
discussed on discord, should be fine to use ambr api for images